src/clj/rems/application/commands.clj

```diff
@@ -341,9 +351,14 @@
                         [event]))))))
 
 (defmethod command-handler :application.command/save-draft
-  [cmd _application _injections]
-  (ok {:event/type :application.event/draft-saved
-       :application/field-values (:field-values cmd)}))
+  [cmd application _injections]
+  (let [forms (:application/forms application)
+        answers (:field-values cmd)
+        answers-formatted (build-index {:keys [:form :field] :value-fn :value} answers)
```

Comment:
you could consider doing `build-index` inside `enrich-form-answers`

Reply:
Done!